server/game/basecard.js

```diff
@@ -184,7 +184,19 @@ class BaseCard {
      * is both in play and not blank.
      */
     persistentEffect(properties) {
-        this.abilities.persistentEffects.push(_.extend({ duration: 'persistent' }, properties));
+        const allowedLocations = ['active plot', 'agenda', 'any', 'play area'];
+        const defaultLocationForType = {
+            agenda: 'agenda',
+            plot: 'active plot'
+        };
+
+        let location = properties.location || defaultLocationForType[this.getType()] || 'play area';
+
+        if(!allowedLocations.includes(location)) {
+            throw new Error(`'${location}' is not a supported effect location.`);
```

Comment:
Yup. I think this is the first place in the server code base but I used it for error messages in the test framework. They're handy.
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals